### DISCUSSION TITLE
Mock plugin not functioning even after caldera server restart

### DISCUSSION THREAD

**Discussion by DulakshiV** [mitre/caldera#2153] [Q&A] [answered]
I logged in as 'red' user on caldera server and pressed the 'enable' button under Advanced-> Configurations -> Plugins -> mock.
Then logged out and closed the terminal where caldera server starting script was running to terminate the server.
Then started the server again. However, when I checked the mock plugin is still not enabled. What could be the reason for this?

Is there a proper command/process to terminate the server rather than just forcefully closing the terminal which ran the server.py? 
Should I have edited the conf/agents.yml file to include agents for the mock plugin?

Version: Caldera 3.1.0, Ubuntu 20.04

**Comment by uruwhy** [comment, marked as answer]
Just to double check:
- Do you have the `mock` plugin installed under `plugins/mock`?
- When you enable the plugin via the GUI and then kill the caldera server, does your configuration file (either `conf/default.yml` or `conf/local.yml`) have the `mock` entry under plugins?
- Lastly, when you restart the server after enable the plugin, does the server debugging output show that the plugin was enabled or if there was an error enabling the plugin? A successful message would look something like:
```
2021-07-27 10:58:01 - INFO  (app_svc.py:107 load) Enabled plugin: mock
```

**Reply by uruwhy** [reply]
Actually, it looks like the issue is in the way you terminated the server. Closing the terminal window force-kills the server process without allowing it to initiate its shut-down sequence, which is when it writes changes to the configuration file. The server should be shutdown using the `ctrl+c` keyboard sequence in the appropriate terminal window, since `server.py` waits for the keyboard interrupt before gracefully shutting down.
